python/src/specifications/specifications_impl.py

```diff
@@ -21,83 +32,106 @@ def spec_csv(architecture_file: str, output_directory: str) -> PluginExecutionRe
-
-    with plugin_result(plugin_name, get_csv) as result:
-        return result
-
-
-def _get_parsed_models(architecture_file: str) -> List[Definition]:
-    with validated_source(architecture_file) as result:
-        return result.definitions
-
-
-def _get_requirements(spec: Definition) -> List[dict]:
-    ret_val: List[dict] = []
-    if spec.get_root_key() in ["spec"]:  # make sure we're actually working with a spec here
-        # collect data
-        spec_dict: dict = spec.structure
-        spec_name = spec_dict["spec"]["name"]
-
-        # handle the spec root requirements
-        if "requirements" in spec_dict["spec"].keys():
-            for req in spec_dict["spec"]["requirements"]:
-                ret_val.append(_gen_spec_line_from_req_dict(spec_name, "", req))
-
-        # handle the requirements in each section
-        if "sections" in spec_dict["spec"].keys():
-            for section in spec_dict["spec"]["sections"]:
-                section_name = section["name"]
-                for req in section["requirements"]:
-                    ret_val.append(_gen_spec_line_from_req_dict(spec_name, section_name, req))
+    reqs = {}
+    req_specs = {}
+    parsed_file = parse(architecture_file)
+    for spec in parsed_file:
+        definition = spec
+        if definition.get_root_key() == "req":
+            reqs[definition.name] = definition
+        if definition.get_root_key() == "req_spec":
+            req_specs[definition.name] = definition
+
+    rows = {}
+    for spec in req_specs:
+        ret_val = _create_rows(req_specs, reqs, spec)
+        rows[spec] = ret_val
+    # rows is a dictionary of lists(each file) of dictionaries(the rows in each file)
+    (success, message) = parent_child_check(rows, reqs)
+    if success is False:
+        return ExecutionResult(plugin_name, "Specifications", ExecutionStatus.GENERAL_FAILURE, message)
+
+    field_names = ["Spec Name", "Section", "ID", "Requirement", "Parents", "Children"]
+
+    # just in case, let's make sure the output directory exists
+    if not path.lexists(output_directory):
+        makedirs(output_directory)
+
+    file_counter = 0
+    for spec_name in req_specs:
+        file_name = spec_name + ".csv"
+        file_name = file_name.replace(" ", "_")
+        output_path = path.join(output_directory, file_name)
+        with open(output_path, "w") as output:
+            writer = csv.DictWriter(output, fieldnames=field_names)
+            writer.writeheader()
+            writer.writerows(rows[spec_name])
+            file_counter = file_counter + 1
+    status = ExecutionStatus.SUCCESS
+    messages: list[ExecutionMessage] = []
+    messages.append(ExecutionMessage(f"{file_counter} CSV spec files written to {output_directory}", MessageLevel.INFO, None, None))
+
+    return ExecutionResult(plugin_name, "Specifications", status, messages)
+
+
+def _create_rows(req_specs, reqs, spec) -> List[dict]:
```

Comment:
Missing docstrings

Reply:
Missing your return, blank line following initial description, and there shouldn't be blank lines between the argument entries.

https://google.github.io/styleguide/pyguide.html#383-functions-and-methods